### ISSUE TITLE
Tests don't run on clean master checkout

### ISSUE THREAD

**Issue by EmilStenstrom** [allenai/allennlp#3570]
**Describe the bug**
When cloning the allennlp repository an running `pytest` i get test errors. This makes it harder to contribute since I'm not sure if I broke something or not. 

<details>
<summary>[Full output of `pytest -vv`]</summary>

```
================================================================================= FAILURES ==================================================================================
__________________________________________________________ TestPosTagIndexer.test_count_vocab_items_uses_pos_tags ___________________________________________________________

self = <allennlp.tests.data.token_indexers.pos_tag_indexer_test.TestPosTagIndexer testMethod=test_count_vocab_items_uses_pos_tags>

    def test_count_vocab_items_uses_pos_tags(self):
        tokens = self.tokenizer.tokenize("This is a sentence.")
        tokens = [Token("<S>")] + [t for t in tokens] + [Token("</S>")]
        indexer = PosTagIndexer()
        counter = defaultdict(lambda: defaultdict(int))
        for token in tokens:
            indexer.count_vocab_items(token, counter)
        assert counter["pos_tokens"] == {"DT": 2, "VBZ": 1, ".": 1, "NN": 1, "NONE": 2}
    
        indexer._coarse_tags = True
        counter = defaultdict(lambda: defaultdict(int))
        for token in tokens:
            indexer.count_vocab_items(token, counter)
>       assert counter["pos_tokens"] == {"AUX": 1, "PUNCT": 1, "DET": 2, "NOUN": 1, "NONE": 2}
E       AssertionError: assert defaultdict(<..., 'PUNCT': 1}) == {'AUX': 1, 'DE...NOUN': 1, ...}
E         Common items:
E         {'DET': 2, 'NONE': 2, 'NOUN': 1, 'PUNCT': 1}
E         Left contains 1 more item:
E         {'VERB': 1}
E         Right contains 1 more item:
E         {'AUX': 1}
E         Full diff:
E         + {'AUX': 1, 'DET': 2, 'NONE': 2, 'NOUN': 1, 'PUNCT': 1}
E         - defaultdict(<class 'int'>,
E         -             {'DET': 2,
E         -              'NONE': 2,
E         -              'NOUN': 1,
E         -              'PUNCT': 1,
E         -              'VERB': 1})

allennlp/tests/data/token_indexers/pos_tag_indexer_test.py:27: AssertionError
--------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------
2020-01-01 12:01:46,737 - INFO - allennlp.common.checks - Pytorch version: 1.3.1
2020-01-01 12:01:46,740 - WARNING - allennlp.data.token_indexers.pos_tag_indexer - Token had no POS tag: <S>
2020-01-01 12:01:46,740 - WARNING - allennlp.data.token_indexers.pos_tag_indexer - Token had no POS tag: </S>
----------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------
12:01:46 - INFO - allennlp.common.checks - Pytorch version: 1.3.1
12:01:46 - WARNING - allennlp.data.token_indexers.pos_tag_indexer - Token had no POS tag: <S>
12:01:46 - WARNING - allennlp.data.token_indexers.pos_tag_indexer - Token had no POS tag: </S>
__________________________________________________________ TestPosTagIndexer.test_tokens_to_indices_uses_pos_tags ___________________________________________________________

self = <allennlp.tests.data.token_indexers.pos_tag_indexer_test.TestPosTagIndexer testMethod=test_tokens_to_indices_uses_pos_tags>

    def test_tokens_to_indices_uses_pos_tags(self):
        tokens = self.tokenizer.tokenize("This is a sentence.")
        tokens = [t for t in tokens] + [Token("</S>")]
        vocab = Vocabulary()
        aux_index = vocab.add_token_to_namespace("AUX", namespace="pos_tags")
        cop_index = vocab.add_token_to_namespace("VBZ", namespace="pos_tags")
        none_index = vocab.add_token_to_namespace("NONE", namespace="pos_tags")
        # Have to add other tokens too, since we're calling `tokens_to_indices` on all of them
        vocab.add_token_to_namespace("DET", namespace="pos_tags")
        vocab.add_token_to_namespace("NOUN", namespace="pos_tags")
        vocab.add_token_to_namespace("PUNCT", namespace="pos_tags")
    
        indexer = PosTagIndexer(namespace="pos_tags", coarse_tags=True)
    
>       indices = indexer.tokens_to_indices(tokens, vocab, "tokens")

allennlp/tests/data/token_indexers/pos_tag_indexer_test.py:43: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
allennlp/data/token_indexers/pos_tag_indexer.py:72: in tokens_to_indices
    return {index_name: [vocabulary.get_token_index(tag, self._namespace) for tag in tags]}
allennlp/data/token_indexers/pos_tag_indexer.py:72: in <listcomp>
    return {index_name: [vocabulary.get_token_index(tag, self._namespace) for tag in tags]}
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Vocabulary with namespaces:  pos_tags, Size: 6 || Non Padded Namespaces: {'*labels', '*tags'}, token = 'VERB', namespace = 'pos_tags'

    def get_token_index(self, token: str, namespace: str = "tokens") -> int:
        if token in self._token_to_index[namespace]:
            return self._token_to_index[namespace][token]
        else:
            try:
>               return self._token_to_index[namespace][self._oov_token]
E               KeyError: '@@UNKNOWN@@'

allennlp/data/vocabulary.py:710: KeyError
--------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------
2020-01-01 12:01:47,061 - INFO - allennlp.common.checks - Pytorch version: 1.3.1
2020-01-01 12:01:47,063 - ERROR - allennlp.data.vocabulary - Namespace: pos_tags
2020-01-01 12:01:47,063 - ERROR - allennlp.data.vocabulary - Token: VERB
--------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------
2020-01-01 12:01:47,063 - ERROR - allennlp.data.vocabulary - Namespace: pos_tags
2020-01-01 12:01:47,063 - ERROR - allennlp.data.vocabulary - Namespace: pos_tags
2020-01-01 12:01:47,063 - ERROR - allennlp.data.vocabulary - Token: VERB
2020-01-01 12:01:47,063 - ERROR - allennlp.data.vocabulary - Token: VERB
----------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------
12:01:47 - INFO - allennlp.common.checks - Pytorch version: 1.3.1
12:01:47 - ERROR - allennlp.data.vocabulary - Namespace: pos_tags
12:01:47 - ERROR - allennlp.data.vocabulary - Token: VERB
______________________________________________________________ TestSrlPredictor.test_prediction_with_no_verbs _______________________________________________________________

self = <allennlp.tests.predictors.srl_test.TestSrlPredictor testMethod=test_prediction_with_no_verbs>

    def test_prediction_with_no_verbs(self):
    
        input1 = {"sentence": "Blah no verb sentence."}
        archive = load_archive(self.FIXTURES_ROOT / "srl" / "serialization" / "model.tar.gz")
        predictor = Predictor.from_archive(archive, "semantic-role-labeling")
        result = predictor.predict_json(input1)
        assert result == {"words": ["Blah", "no", "verb", "sentence", "."], "verbs": []}
    
        input2 = {"sentence": "This sentence seems to have a verb."}
        results = predictor.predict_batch_json([input1, input2])
        assert results[0] == {"words": ["Blah", "no", "verb", "sentence", "."], "verbs": []}
>       assert results[1] == {
            "verbs": [
                {
                    "verb": "seems",
                    "description": "This sentence seems to have a verb .",
                    "tags": ["O", "O", "O", "O", "O", "O", "O", "O"],
                }
            ],
            "words": ["This", "sentence", "seems", "to", "have", "a", "verb", "."],
        }
E       AssertionError: assert {'verbs': [{'...e', 'a', ...]} == {'verbs': [{'d...e', 'a', ...]}
E         Common items:
E         {'words': ['This', 'sentence', 'seems', 'to', 'have', 'a', 'verb', '.']}
E         Differing items:
E         {'verbs': [{'description': 'This sentence seems to have a verb .', 'tags': ['O', 'O', 'O', 'O', 'O', 'O', ...], 'verb'... {'description': 'This sentence seems to have a verb .', 'tags': ['O', 'O', 'O', 'O', 'O', 'O', ...], 'verb': 'have'}]} != {'verbs': [{'description': 'This sentence seems to have a verb .', 'tags': ['O', 'O', 'O', 'O', 'O', 'O', ...], 'verb': 'seems'}]}
E         Full diff:
E         {'verbs': [{'description': 'This sentence seems to have a verb .',
E         'tags': ['O', 'O', 'O', 'O', 'O', 'O', 'O', 'O'],
E         -             'verb': 'seems'},
E         +             'verb': 'seems'}],
E         ?                             +
E         -            {'description': 'This sentence seems to have a verb .',
E         -             'tags': ['O', 'O', 'O', 'O', 'O', 'O', 'O', 'O'],
E         -             'verb': 'have'}],
E         'words': ['This', 'sentence', 'seems', 'to', 'have', 'a', 'verb', '.']}

allennlp/tests/predictors/srl_test.py:94: AssertionError

```

</details>

**To Reproduce**
Steps to reproduce the behavior
1. Run `pytest` on a clean checkout and wait for tests to fail

**Expected behavior**
All tests pass on clean checkout.

**System (please complete the following information):**
 - OS: OSX
 - Python version: Python 3.6.8
 - AllenNLP version: "I installed from master"
 - PyTorch version: "Installed with pip install -r requirements.txt"

**Additional context**

**Comment by schmmd** [comment]
We run `allennlp test-install` on every commit to master so this has something to do with the environment.  I'm attempting to reproduce this on my mac.

**Comment by matt-gardner** [comment]
It's due to an updated spacy model, I believe.  We should move away from tests that fail when this happens.

**Comment by matt-gardner** [comment]
It's also now happening in CI on PR commits.

**Comment by schmmd** [comment]
Seems like we should `@pytest.mark.skip` these tests for now.  I wasn't able to reproduce this locally because I ran into different errors in a fresh conda environment ;-(

```
allennlp/tests/data/dataset_readers/multiprocess_dataset_reader_test.py .Fatal Python error: Aborted

Current thread 0x0000000109ac2dc0 (most recent call first):
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/multiprocessing/popen_fork.py", line 70 in _launch
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/multiprocessing/popen_fork.py", line 20 in __init__
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/multiprocessing/context.py", line 277 in _Popen
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/multiprocessing/context.py", line 223 in _Popen
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/multiprocessing/process.py", line 112 in start
  File "/Users/michael/hack/allenai/allennlp/allennlp/data/dataset_readers/multiprocess_dataset_reader.py", line 156 in _instances
  File "/Users/michael/hack/allenai/allennlp/allennlp/tests/data/dataset_readers/multiprocess_dataset_reader_test.py", line 60 in test_multiprocess_read
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/unittest/case.py", line 628 in run
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/unittest/case.py", line 676 in __call__
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/unittest.py", line 207 in runtest
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/runner.py", line 131 in pytest_runtest_call
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/runner.py", line 207 in <lambda>
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/runner.py", line 234 in from_call
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/runner.py", line 207 in call_runtest_hook
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/flaky/flaky_pytest_plugin.py", line 129 in call_and_report
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/runner.py", line 96 in runtestprotocol
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/runner.py", line 81 in pytest_runtest_protocol
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/flaky/flaky_pytest_plugin.py", line 92 in pytest_runtest_protocol
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/main.py", line 270 in pytest_runtestloop
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/main.py", line 246 in _main
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/main.py", line 196 in wrap_session
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/main.py", line 239 in pytest_cmdline_main
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/michael/miniconda3/envs/allennlp/lib/python3.7/site-packages/_pytest/config/__init__.py", line 92 in main
  File "/Users/michael/hack/allenai/allennlp/allennlp/commands/test_install.py", line 75 in _run_test
  File "/Users/michael/hack/allenai/allennlp/allennlp/commands/__init__.py", line 102 in main
  File "/Users/michael/hack/allenai/allennlp/allennlp/run.py", line 18 in run
  File "/Users/michael/miniconda3/envs/allennlp/bin/allennlp", line 11 in <module>
^C[INFO/SyncManager-4] process exiting with exitcode 0
```

**Comment by DeNeutoy** [comment]
Fixed by #3619
